great_expectations/expectations/metrics/util.py

```diff
@@ -604,6 +611,93 @@ def column_reflection_fallback(
     return col_info_dict_list
 
 
+def get_typed_column_names(
```

Comment:
Love the solution, but something about this name doesn't sit right with me. I don't think the column names are "typed". Maybe `get_case_sensitive_column_names`?

Reply:
Upon second thought this handles spaces as well. `get_existing_column_names` might work since this doesn't do anything with quoting for non-SQL Alchemy.

Reply:
@NathanFarmer Changed to `get_dbms_compatible_column_names()` -- how about that?!